tests/ttnn/unit_tests/operations/test_moreh_softmin.py

```diff
@@ -397,3 +389,84 @@ def test_softmin_backward_optional_output_tensor(shape_dim, optional_output_tens
     passing, out = comp_allclose_and_pcc(x.grad, tt_dev, rtol=rtol, atol=atol)
     logger.info(out)
     assert passing
+
+
+@pytest.mark.parametrize(
+    "shape_dim",
+    [
+        [[32, 32], 1],  # H small
+        [[32, 32], 0],  # W small
+        [[2, 3, 32 * 4, 32 * 5], 3],  # H large
+        [[2, 3, 32 * 4, 32 * 5], 2],  # W large,
+        [[1, 15, 32, 32], 1],  # C
+    ],
+)
+def test_softmin_callback(shape_dim, device, use_program_cache):
+    shape, dim = shape_dim
+    torch.manual_seed(0)
+
+    x = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16)
+
+    dev_x = ttnn.Tensor(x, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
+
+    tt_cpu = F.softmin(x, dim)
+    for i in range(2):
+        dev_x = ttnn.Tensor(x, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
+        tt_npu = ttnn.operations.moreh.softmin(dev_x, dim)
+        if i == 0:
+            num_program_cache_entries = device.num_program_cache_entries()
+            assert num_program_cache_entries > 0
+        else:
+            assert device.num_program_cache_entries() == num_program_cache_entries
+        torch_dummy = torch.randn([32, 32])
+        tt_dummy = to_npu(torch_dummy, device)
+
+    assert list(tt_npu.shape.with_tile_padding()) == list(tt_cpu.shape)
+    tt_dev = tt_npu.cpu().to(ttnn.ROW_MAJOR_LAYOUT).to_torch().to(torch.bfloat16)
+
+    rtol = atol = 0.05
+    passing, out = comp_allclose_and_pcc(tt_cpu, tt_dev, rtol=rtol, atol=atol)
+    logger.debug(out)
+    assert passing
+
+
+@pytest.mark.parametrize(
+    "shape_dim",
+    [
+        [[32, 32], 1],  # H small
+        [[32, 32], 0],  # W small
+        [[2, 3, 32 * 4, 32 * 5], 3],  # H large
+        [[2, 3, 32 * 4, 32 * 5], 2],  # W small
+        [[1, 15, 32, 32], 1],  # C
+    ],
+)
+def test_softmin_backward_callback(shape_dim, device, use_program_cache):
+    shape, dim = shape_dim
+    torch.manual_seed(0)
+
+    x = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16).requires_grad_(True)
+
+    y = F.softmin(x, dim)
+    dev_y = ttnn.Tensor(y, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
+
+    dy = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16)
+    dev_dy = ttnn.Tensor(dy, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
+
+    y.backward(dy)
+    for i in range(2):
+        tt_npu = ttnn.operations.moreh.softmin_backward(dev_y, dev_dy, dim)
+        if i == 0:
```

Comment:
This test will pass even if the input update is missing from the callback because dev_y and dev_dy remain the same in both the first and second loops. 
This could lead to issues with the test's reliability.

Reply:
Does this work?
`for i in range(2):
        x = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16).requires_grad_(True) + i

        y = F.log_softmax(x, dim)
        dev_y = ttnn.Tensor(y, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)

        dy = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16)
        dev_dy = ttnn.Tensor(dy, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)

        y.backward(dy)
        tt_npu = ttnn.operations.moreh.logsoftmax_backward(dev_y, dev_dy, dim, strategy=strategy)
        if i == 0:
            num_program_cache_entries = device.num_program_cache_entries()
            assert num_program_cache_entries > 0
        else:
            assert device.num_program_cache_entries() == num_program_cache_entries
        torch_dummy = torch.randn([32, 32])
        tt_dummy = to_npu(torch_dummy, device)
`

Reply:
Yes, that sounds good.
Because when the second loop starts, PyTorch creates a new `x` tensor, causing the value to change.
` x = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16).requires_grad_(True) + i`
Since a new value is generated without the need for + i, the variable i is not necessary.